### PR TITLE
RFE 426 overlay for Planetary Conditions

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
RFE #426 overlay for Planetary Conditions

cloned the KeyBindingsOverlay to display Planetary Conditions

Are there any other conditions that would be good to add to this?

image

### CODE REVIEW

**Comment by SJuliez** [comment]
Also: I remember that people had difficulty with this initially because the keybind didn't work properly. iirc you'd have to add the keybind default to the keybinds.xml file that is somewhere in the config folder.

**Comment by SJuliez** [comment]
Thirdly :) you might wanna make sure that it doesnt show up in the board editor.

**Comment by kuronekochomusuke** [comment]
- colors can be set in advance options
- added condition indicators
- change temperature color based on extreme temperature ranges
- align to right side

image

**Comment by SJuliez** [comment]
> Also: I remember that people had difficulty with this initially because the keybind didn't work properly. iirc you'd have to add the keybind default to the keybinds.xml file that is somewhere in the config folder.

Sorry if I was not making sense: I was speaking of the key bind overlay. But the same is true for the new overlay.

**Comment by kuronekochomusuke** [comment]
I tried updating the keybinds.xml for  this and added in null checks for the  board editor

**Comment by kuronekochomusuke** [comment]
- add advance option to hide default conditions

**Comment by kuronekochomusuke** [comment]
- scale overlay based on the GUI scale

image

image

**Comment by HammerGS** [comment]
Looking good!

**Comment by kuronekochomusuke** [comment]
- add advance setting to show the header, labels, values, and indicators

image

indicatorScreenshot 2022-11-11 101335

**Comment by kuronekochomusuke** [comment]
This is now ready for review

File: megamek/src/megamek/client/ui/swing/boardview/PlanetaryConditionsOverlay.java [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        @Override
+        public void gameTurnChange(GameTurnChangeEvent e) {
+            // The active player has changed
+            changed = true;
+        }
+    };
+
+    @Override
+    public void preferenceChange(PreferenceChangeEvent e) {
+        if (e.getName().equals(KeyBindParser.KEYBINDS_CHANGED)) {
```
Comment (SJuliez) [inline megamek/src/megamek/client/ui/swing/boardview/PlanetaryConditionsOverlay.java:397]: I believe it's not hurtful to just set changed to true here without doing any checks (PreferenceChangeEvents don't happen all the time); this way it'll react to changes to its advanced settings.

**Review by SJuliez — CHANGES_REQUESTED** [review]
I like this a lot! Just one small request then I'll merge it.

**Review by SJuliez — APPROVED** [review]

**Pull request merged**